src/go/rdctl/pkg/runner/queue.go

```diff
@@ -0,0 +1,63 @@
+package runner
+
+import (
+	"context"
+	"errors"
+)
+
+var ErrContextDone error = errors.New("context marked done")
+
+// TaskRunner accepts functions and asynchronously calls them in the
+// order they were received. Before each function is called, TaskRunner
+// checks whether its context is marked done; if so, it stops calling
+// functions.
```

Comment:
So, this is like a `golang.org/x/sync/errgroup.Group` that executes tasks in order. (I'm just trying to sort out my thoughts; nothing needs to be changed here.)